tests/components/test_proximity.py

```diff
@@ -7,21 +7,89 @@
 
 import homeassistant.core as ha
 from homeassistant.components import proximity
+from homeassistant.components import zone
+import homeassistant.components.device_tracker as device_tracker
+import homeassistant.util.dt as dt_util
+from datetime import datetime, timedelta
 
 class TestProximity:
     """ Test the Proximity component. """
 
     def setup_method(self, method):
         self.hass = ha.HomeAssistant()
+        
+        self.yaml_devices = self.hass.config.path(device_tracker.YAML_DEVICES)
+        
+        assert zone.setup(self.hass, {
+            'zone': [
+                {
+                    'name': 'home',
+                    'latitude': 2.1,
+                    'longitude': 1.1,
+                    'radius': 10,
+                },
+                {
+                    'name': 'work',
+                    'latitude': 100,
+                    'longitude': 100,
+                    'radius': 10,
+                },
+            ]
+        })
+        self.hass.pool.block_till_done()
```

Comment:
Everything in the setup method is run before every single test. You're testing actual logic in the setup method which you shouldn't. Also you don't have to block till done.